diffprivlib/models/tree.py

```diff
@@ -0,0 +1,602 @@
+            for value in feature_domains[str(split_feature)]:
+                child_node = self._build([f for f in features if f != split_feature], feature_domains, current_depth+1)
+                node.add_cat_child(value, child_node)
+        else:
+            node.set_split_type(DecisionNode.CONT_SPLIT)
+            split_value = np.random.uniform(feature_domains[str(split_feature)][0],
+                                            feature_domains[str(split_feature)][1])
+            node.set_split_value(split_value)
+            left_domain = {k: v if k != str(split_feature) else [v[0], split_value]
+                           for k, v in feature_domains.items()}
+            right_domain = {k: v if k != str(split_feature) else [split_value, v[1]]
+                            for k, v in feature_domains.items()}
+            left_child = self._build(features, left_domain, current_depth+1)
+            right_child = self._build(features, right_domain, current_depth+1)
+            node.set_left_child(left_child)
+            node.set_right_child(right_child)
+
+        return node
+
+    def fit(self, X, y):
+        if not isinstance(self._cat_feature_threshold, numbers.Integral) or self._cat_feature_threshold < 0:
+            raise ValueError('Categorical feature threshold should be a positive integer;'
+                             f'got {self._cat_feature_threshold}')
+
+        X, y = np.array(X), np.array(y)
+        check_X_y(X, y)
+
+        if self._cat_features is None:
+            self._cat_features = get_cat_features(X, self._cat_feature_threshold)
+
+        if self._feature_domains is None:
+            warnings.warn(
+                "feature_domains parameter hasn't been specified, so falling back to determining domains from the data.\n"
+                "This may result in additional privacy leakage. To ensure differential privacy with no "
+                "additional privacy loss, specify `feature_domains` according to the documentation",
+                PrivacyLeakWarning)
+            self._feature_domains = get_feature_domains(X, self._cat_features)
+
+        if self._classes is None:
+            self._classes = np.unique(y)
+
+        n_features = X.shape[1]
+        features = list(range(n_features))
+
+        self._root = self._build(features, self._feature_domains)
+
+        for i in range(len(X)):
+            node = self._root.classify(X[i])
+            node.update_class_count(y[i])
+
+        self._root.set_noisy_label(self._epsilon, self._classes)
+        self._fitted = True
+
+        return self
+
+    def predict(self, X):
+        if not self.check_is_fitted():
+            raise Exception('Model is not fitted yet')
+
+        y = []
+        X = np.array(X)
+        check_array(X)
+
+        for x in X:
+            node = self._root.classify(x)
+            y.append(node.noisy_label)
+
+        return np.array(y)
+
+    def check_is_fitted(self):
```

Comment:
Let's use `check_is_fitted` from `sklearn.utils.validation` instead, which can be done one we refactor the fitted attribute names as described above.